checkrole.yml.sample

```
@@ -0,0 +1,30 @@
+---
+
+# check host hardware specification against a given set of Ceph roles.
+#
+# Requires the variables to be passed in by the caller or vars defined
+# e.g.
+# vars:
+#   inventory:
+#     host1: osds,mons
+#     host2: rgws
+#     host3: mons
+#   mode: prod
+#   deployment: container
+#   osdtype: bluestore
+#   flashusage: journal 
+
+- name: Validate hosts against desired cluster state
+  hosts: all
+  become: true
+  gather_facts: false
+
+  tasks:
+  
+    - name: CEPH_CHECK_ROLE
+      ceph_check_role:
+        role: "{{ inventory[inventory_hostname] }}"
```

with this variable, it feels to me it makes us to rewrite the inventory
I would probably suggest something like following:
```suggestion
        role: "{{ ','.join(hostvars[inventory_hostname]['group_names']) }}"
```
to avoid having to rewrite the inventory
or :
```suggestion
        role: "{{ inventory[inventory_hostname] | default(','.join(hostvars[inventory_hostname]['group_names'])) }}"
```
if you really want to keep the possibility to use that `inventory` variable

I even wonder whether you could manage with this param directly in the module so it wouldn't be a required param eg :
```python
    fields = dict(
        role=dict(
            type='str',
            required=False),
```